### PR TITLE
Validate if build section has image on vanilla clusters

### LINKED ISSUE #2920: Validate if build section has image on vanilla clusters
**Describe the bug**
On vanilla clusters we can't infer the name of the images built. If the manifest validation detects that the user is running an okteto command inside a vanilla cluster with a build without image name we should guide the user to update their manifest. 

**To Reproduce**

1. Copy the following `okteto manifest`
```yaml
build:
  autocreate:
    context: .
dev:
  autocreate:
    sync:
      - .:/src
    autocreate: true
```
2. Run okteto up on a vanilla cluster
3. See that the error is not descriptive

**Expected behavior**
I want a message guiding the users to update their manifest to include a image field on the build section
 
**Desktop (please complete the following information):**

- Version 2.4.2

### PR DESCRIPTION
**Pull request by aniruddha2000**
Signed-off-by: Aniruddha Basak <codewithaniruddha@gmail.com>

# Proposed changes

Fixes #2920

This PR includes the validations to check for image in the build section on a vanilla cluster.

### CODE REVIEW

File: pkg/model/manifest.go
Comment (abs007) [inline pkg/model/manifest.go:676]: Probable nitpicking but you'd want to change the string to:
```go
"Error getting the image name for the service '%s'. Please update the manifest to include an image field in the build section when using a cluster that doesn't have Okteto installed"
```

Just some grammar fixes fyi

Comment (jLopezbarb) [reply pkg/model/manifest.go:676]: Yes, I like @abs007 message. Thanks!!

**Review by jLopezbarb — APPROVED** [review]

**Comment by aniruddha2000** [comment]
@jLopezbarb any scope for adding a unit test?

**Comment by aniruddha2000** [comment]
@jLopezbarb can you let me know which `isOkteto()` you are referring?

**Comment by jLopezbarb** [comment]
> @jLopezbarb can you let me know which `isOkteto()` you are referring?

Hi @aniruddha2000. I'm referring to [this isOkteto()](https://github.com/okteto/okteto/blob/master/pkg/okteto/context.go#L198). It should be fixed by https://github.com/okteto/okteto/pull/3130 but we need to review if it's the best approach or not. Hopefully it will be merged at the end of the week and will unblock your PR

**Comment by jLopezbarb** [comment]
Hi @aniruddha2000, we have merged the PR that was blocking this PR, can you update your branch with master and use the function we commented above?

**Comment by aniruddha2000** [comment]
Hi @jLopezbarb does `IsOkteto()` returns back bool whether the cluster has okteto installed or not?
It would be nice if the function had a one-liner description.

**Comment by jLopezbarb** [comment]
> Hi @jLopezbarb does `IsOkteto()` returns back bool whether the cluster has okteto installed or not?
> It would be nice if the function had a one-liner description.

@aniruddha2000 yes, Could you add the comment in the function please? Thanks in advance

**Comment by aniruddha2000** [comment]
@jLopezbarb Hi, when I am trying to import the `"github.com/okteto/okteto/pkg/okteto"` package it's breaking some of the files in the project are facing import cycle issues as well.

```bash
go build -trimpath -ldflags "-s -w -X github.com/okteto/okteto/pkg/config.VersionString=1b7e61a9" -tags "osusergo netgo static_build" -o /home/aniruddha/Desktop/k8s/projects/okteto/bin/okteto
package github.com/okteto/okteto
        imports github.com/okteto/okteto/cmd
        imports github.com/okteto/okteto/cmd/build/v1
        imports github.com/okteto/okteto/cmd/utils
        imports github.com/okteto/okteto/pkg/k8s/apps
        imports github.com/okteto/okteto/pkg/k8s/deployments
        imports github.com/okteto/okteto/pkg/model
        imports github.com/okteto/okteto/pkg/okteto
        imports github.com/okteto/okteto/pkg/k8s/ingresses
        imports github.com/okteto/okteto/pkg/model: import cycle not allowed
make: *** [Makefile:72: build] Error 1
```

**Comment by AgustinRamiroDiaz** [comment]
> @jLopezbarb Hi, when I am trying to import the `"github.com/okteto/okteto/pkg/okteto"` package it's breaking some of the files in the project are facing import cycle issues as well.

@aniruddha2000 have you merged the latest changes from master into your branch?

**Comment by aniruddha2000** [comment]
@AgustinRamiroDiaz yes I got the latest changes from the okteto `master` branch

```bash
1b7e61a9 (HEAD -> aniruddha/validate-build-image-vanila-cluster) Add manifest validation to include image on a vanilla cluster
2cd722e7 (origin/master, origin/HEAD, master) fix: add error on init container volume full (#3189)
1a94bc98 Module refactor (#3130)
4bb83c9e feat: add specific timeout to dependencies (#3190)
e2832233 Support both string and sequence of string in a manifest's `cache_from` property (#3162)
6bd85417 Updated error messaging when ctx use/update is used for vanilla clusters (#3191)
2c4ae3a6 logs: add logs to syncthing terminating process (#3184)
```

**Comment by AgustinRamiroDiaz** [comment]
Hey @abs007! Thank you again for your contribution :heart: 

@jLopezbarb and I were giving this some thoughts and tried some code changes, but the dependencies around `okteto.IsOkteto()` and the "okteto context" are really coupled to most of the code. We couldn't see a quick workaround for this to unlock you :disappointed:. 

Feel free to search for other issues in our backlog :+1:

**Comment by abs007** [comment]
Happy to help, though I just did a small part. Majority of the discussion was carried out by @aniruddha2000

**Comment by aniruddha2000** [comment]
@jLopezbarb @AgustinRamiroDiaz Sure!
If anything comes into the picture do discuss it in slack I am happy to add my opinions to this feature.

Again thanks to @abs007 for your efforts as well!

**Pull request closed without merging**
